src/Elastic.Markdown/Myst/Components/ApplicableToComponent.cshtml

```diff
@@ -111,209 +111,212 @@
+				ProductLifecycle.Removed => "Removed",
+				_ => ""
+			};
+			
+			var realVersion = TryGetRealVersion(applicability, out var v) ? v : null;
+
+			if (realVersion is not null)
 			{
-				case ProductLifecycle.TechnicalPreview:
-					if (TryGetRealVersion(applicability, out var previewVersion) && previewVersion > versioningSystem.Current)
-					{
-						badgeText = "Planned";
-						lifecycleTooltip = "We plan to add this functionality in a future update. Plans may change without notice.";
-						lifecycleClass = "planned";
-					}
-					else
-					{
-						lifecycleTooltip = "This functionality is in technical preview and may be changed or removed in a future release. Elastic will work to fix any issues, but features in technical preview are not subject to the support SLA of official GA features.";
-					}
 
-					break;
-				case ProductLifecycle.Beta:
-					if (TryGetRealVersion(applicability, out var betaVersion) && betaVersion > versioningSystem.Current)
-					{
-						badgeText = "Planned";
-						lifecycleTooltip = "We plan to add this functionality in a future update. Plans may change without notice.";
-						lifecycleClass = "planned";
-					}
-					else
+				if (realVersion <= versioningSystem.Current)
+				{
+					tooltipText = $"{lifecycleFull} in {keyFull} in version {realVersion} and later unless otherwise specified.";
+				}
+				else
+				{
+					if (applicability.Lifecycle is ProductLifecycle.GenerallyAvailable or ProductLifecycle.Beta or ProductLifecycle.TechnicalPreview)
 					{
-						lifecycleTooltip = "This functionality is in beta and is subject to change. The design and code is less mature than official GA features and is being provided as-is with no warranties. Beta features are not subject to the support SLA of official GA features.";
+						tooltipText = $"We plan to add this in a future {keyFull} update. Plans may change without notice.";
 					}
-
-					break;
-				case ProductLifecycle.GenerallyAvailable:
-					if (TryGetRealVersion(applicability, out var version) && version > versioningSystem.Current)
+					else if (applicability.Lifecycle is ProductLifecycle.Deprecated)
 					{
-						badgeText = "Planned";
-						lifecycleTooltip = "We plan to add this functionality in a future update. Plans may change without notice.";
-						lifecycleClass = "planned";
+						tooltipText = $"We plan to deprecate this in a future {keyFull} update. Plans may change without notice.";
 					}
-
-					break;
-				case ProductLifecycle.Deprecated:
-					if (TryGetRealVersion(applicability, out var deprecatedVersion) && deprecatedVersion > versioningSystem.Current)
+					else if (applicability.Lifecycle is ProductLifecycle.Removed)
 					{
-						badgeText = "Deprecation planned";
-						lifecycleTooltip = "We plan to deprecate this functionality in a future update. Plans may change without notice.";
+						tooltipText = $"We plan to remove this in a future {keyFull} update. Plans may change without notice.";
```

Comment:
>"We plan to remove this in a future {keyFull} update. Plans may change without notice.";

Maybe this instead?

"Removal planned for future {keyFull} release. Plans may change without notice.";